scripts/pull.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+# (C) 2022 GoodData Corporation
+import os, shutil, tempfile
+from gooddata_sdk import GoodDataSdk
+from pathlib import Path
+from dotenv import load_dotenv
+
+# TODO "gooddata_layouts" should be imported from the lib
+LAYOUT_ROOT_FOLDER = "gooddata_layouts"
+
+def main():
+    # Get current environment, defaulting to "development"
+    environment = os.getenv("GD_ENV", "development")
+
+    # Load corresponding env variables from file, if exists
+    load_dotenv(f".env.{environment}")
+
+    # Get all the necessary arguments from environment
+    host = os.getenv("GD_HOST")
+    if host is None:
+        raise RuntimeError("GD_HOST environmental variable must be defined")
+
+    token = os.getenv("GD_TOKEN")
+    if token is None:
+        raise RuntimeError("GD_TOKEN environmental variable must be defined")
+
+    layout_path = Path.cwd() / LAYOUT_ROOT_FOLDER
+
+    sdk = GoodDataSdk.create(host, token)
+
+    # Make sure the GoodData server is running
+    if not sdk.support.is_available:
+        raise RuntimeError(f"GoodData server at {host} is unavailable")
```

Comment:
Think if it would make sense to have something like GoodDataSdk wrapper class with the following content:
```
class GoodDataSdkWrapper:
    host = get_mandatory_env_var("GD_HOST")

    sdk = GoodDataSdk.create(host, get_mandatory_env_var("GD_TOKEN"))
    if not sdk.support.is_available:
        raise RuntimeError(f"GoodData server at {host} is unavailable")
```

When PythonSDK 1.3 is released, it would be better to use the `.create_from_config` method to make an instance of GoodDataSdk.

Reply:
I have some doubts as to how far should we go with keeping it DRY. This are supposed to be super easy to read and self-sufficient code samples.

If we do want to move this code, I would choose to create a factory function rather than the whole class. Something like this, maybe?

```python
# utils.py:
import os
from gooddata_sdk import GoodDataSdk

def get_mandatory_env_var(key: str) -> str:
    try:
        return os.environ[key]
    except KeyError:
        raise RuntimeError(f"{key} environmental variable must be defined")

def get_gooddata_sdk() -> GoodDataSdk:
   host = get_mandatory_env_var("GD_HOST")
   token = get_mandatory_env_var("GD_TOKEN")

   sdk = GoodDataSdk.create(host, token)

   # Make sure the GoodData server is running
   if not sdk.support.is_available:
       raise RuntimeError(f"GoodData server at {host} is unavailable")

   return sdk
```

What do you think?

Reply:
The factory also sounds reasonable.